### PR TITLE
Deprecating several backend menu related classes for 4.0

### PR DESCRIPTION
**Pull request by Hackwar**
This PR deprecates a bunch of classes and methods for #23452

### CODE REVIEW

File: administrator/modules/mod_menu/helper.php [hunk 1]
```diff
@@ -22,6 +22,8 @@ abstract class ModMenuHelper
 	 * @return  array  An array of the available menus (from the menu types table).
 	 *
 	 * @since   1.6
+	 *
+	 * @deprecated  4.0
```
Comment (HLeithner) [inline administrator/modules/mod_menu/helper.php:26]: No replacement?

File: libraries/src/Menu/Tree.php [hunk 2]
```diff
@@ -167,6 +182,8 @@ public function addChild(Node $node, $setCurrent = false)
 	 * @return  string	CSS class name
 	 *
 	 * @since   3.8.0
+	 *
+	 * @deprecated  4.0
```
Comment (HLeithner) [inline libraries/src/Menu/Tree.php:186]: No replacement?

File: libraries/src/Menu/Tree.php [hunk 3]
```diff
@@ -211,6 +228,8 @@ public function getIconClass()
 	 * @return  string[]
 	 *
 	 * @since   3.8.0
+	 *
+	 * @deprecated  4.0
```
Comment (HLeithner) [inline libraries/src/Menu/Tree.php:232]: No replacement?

File: administrator/modules/mod_menu/helper.php (hunk 1 above)
Comment (Hackwar) [reply administrator/modules/mod_menu/helper.php:26]: This is not used in 3.9 already anymore and has been replaced with a completely different system. This deprecation notice should have been added wherever the flexible backend menu was added.

File: libraries/src/Menu/Tree.php (hunk 2 above)
Comment (Hackwar) [reply libraries/src/Menu/Tree.php:186]: Again something that is not necessary in the system. If you need something like this, write it yourself. It's 5 lines of code.

File: libraries/src/Menu/Tree.php (hunk 3 above)
Comment (Hackwar) [reply libraries/src/Menu/Tree.php:232]: This has no usefull purpose in 3.9 already. Tell me what it is used for and I remove the notice. It is not used in the menu system.

**Comment by HLeithner** [comment]
thx

**Pull request merged**